Update readme and changelog for preview 3 release (#10672)
Azure/azure-sdk-for-java: sdk/search/azure-search-documents/CHANGELOG.md

# Release History

## 1.0.0-beta.3 (2020-05-05)

> - Replaced `isRetrievable` API to `isHidden`. Boolean value flips over from `retrievable` to `hidden`.

Reviewer: ```suggestion
- Replaced `isRetrievable` API with `isHidden`, parameter name changed from `retrievable` to `hidden`.
```

> - Changed search version from `2019-05-06` to `2019-05-06-Preview`

Reviewer: ```suggestion
- Changed Azure Search service version from `2019-05-06` to `2019-05-06-Preview`
```

> - Changed `createOrUpdate*` and `delete*` APIs in `SearchServiceClient` to use boolean `onlyIfUnchanged` instead of `MatchConditioans`.

Reviewer: ```suggestion
- Changed `createOrUpdate` and `delete` APIs in `SearchServiceClient` to use boolean `onlyIfUnchanged` instead of `MatchConditions`.
```

> - Updated reactor core to `3.3.5.RELEASE` and reactor netty to `0.9.7.0.9.7.RELEASE`

Reviewer: Shouldn't need to include this here as this is coming through from Azure Core and the Netty module. If you do want to include it remove the bit about Netty as the clients are HTTP transport agnostic.
